sql/core/src/main/java/org/apache/spark/sql/execution/datasources/parquet/SpecificParquetRecordReaderBase.java

```diff
@@ -199,12 +151,21 @@ public void initialize(InputSplit inputSplit, TaskAttemptContext taskAttemptCont
    */
   protected void initialize(String path, List<String> columns) throws IOException {
     Configuration config = new Configuration();
-    config.set("spark.sql.parquet.binaryAsString", "false");
-    config.set("spark.sql.parquet.int96AsTimestamp", "false");
+    config.setBoolean(SQLConf.PARQUET_BINARY_AS_STRING().key() , false);
+    config.setBoolean(SQLConf.PARQUET_INT96_AS_TIMESTAMP().key(), false);
 
     this.file = new Path(path);
     long length = this.file.getFileSystem(config).getFileStatus(this.file).getLen();
-    ParquetMetadata footer = readFooter(config, file, range(0, length));
+    ParquetReadOptions options = HadoopReadOptions
+      .builder(config)
+      .withRange(0, length)
+      .build();
+
+    ParquetMetadata footer;
+    try (ParquetFileReader reader = ParquetFileReader
```

Comment:
The change seems okay but can we do a microbenchmark to confirm there's no performance impact here? I am still confused why I faced and concluded that using new Parquet API caused performance regression when I tired it by myself a while ago ... (sorry I forgot all about it).

Reply:
Sure, I can do that. I did run `FilterPushdownBenchmark` (see comments above) before and didn't see regression. I'll do that with the new Parquet version.

Reply:
Do we need a benchmark here or is that probably convincing? this would be a good simplifying change but yeah the only real question is whether there are any side effects.

Reply:
@HyukjinKwon @srowen I've run the `FilterPushdownBenchmark` again with this PR and I don't see much difference before and after. I've put the result [here](https://gist.github.com/sunchao/75cc9e966108bce4353818ce1b59d200) in case you are curious.

Reply:
I'm just eyeballing the diff, and most cases are about the same, but there seem to be a number of cases where this brings a 10-20% improvement, like the InSet -> InFilter tests. Seems worthwhile to commit if tests pass, and they seem to.

Reply:
It is possible that I wrote the codes wrongly at that time .. The results look good. I guess it's fine to go ahead too.